### PR TITLE
REP-2825 Bump server versions to 4.4.23, 5.0.19, 6.0.8 and 7.0.0-rc7

### PR DESCRIPTION
**Pull request by edobranov**
Bumps these 3 versions for mongosync's updated minimum server version requirements (cc @ShaneHarvey)

### CODE REVIEW

**Review by ShaneHarvey — APPROVED** [review]

File: .evergreen/download-mongodb.sh [hunk 1]
```diff
@@ -57,11 +57,11 @@ get_mongodb_download_url_for ()
    VERSION_RAPID="6.3.1"
    VERSION_70="7.0.0-rc3"
```
Comment (ShaneHarvey) [inline .evergreen/download-mongodb.sh:58]: Would you mind updating this to r7.0.0-rc7 as well?

Comment (edobranov) [reply .evergreen/download-mongodb.sh:58]: Sure thing! Would that be `r7.0.0-rc7`, or `7.0.0-rc7` without the leading `r`?

Comment (ShaneHarvey) [reply .evergreen/download-mongodb.sh:58]: 7.0.0-rc7 without the leading r

Comment (edobranov) [reply .evergreen/download-mongodb.sh]: Cool, feel free to merge when you can.

**Pull request merged**